### PR TITLE
Grouping Chisel API

### PR DESCRIPTION
**Pull request by azidar**
**Type of change**: feature request

**Impact**: API addition (no impact on existing code)

**Development Phase**: implementation

**Release Notes**
At the request of @colinschmidt, added a chisel annotation API for triggering the `firrtl.transforms.GroupComponents` transformation.

### CODE REVIEW

**Comment by edtechexplorers** [comment]
LGTM! Thanks!

**Comment by colinschmidt** [comment]
LGTM! Thanks!

File: src/main/scala/chisel3/util/experimental/group.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package chisel3.util.experimental
+
+import chisel3._
+import chisel3.core.{ChiselAnnotation, CompileOptions, RunFirrtlTransform, annotate, requireIsHardware}
+import firrtl.Transform
+import firrtl.transforms.{GroupAnnotation, GroupComponents}
+
+
+object group {
+  def apply[T <: Data](components: Seq[T], newModule: String, newInstance: String, outputSuffix: Option[String] = None, inputSuffix: Option[String] = None)(implicit compileOptions: CompileOptions): Unit = {
```
Comment (edwardcwang) [inline src/main/scala/chisel3/util/experimental/group.scala]: Scaladoc would be appreciated :)

**Comment by ducky64** [comment]
Would it make sense for this to be a scoped API? Though I guess the problem there is that the items in the scope aren't 'visible' outside, depending on if they need to be referred to from outside or if it's more modular and encapsulated.

**Comment by colinschmidt** [comment]
The scoping probably won't work for most of my use cases. If I were able to cleanly put a scope around the items I wanted grouped I wouldn't need this pass at all.

**Comment by jwright6323** [comment]
cc @harrisonliew

Is this still in the works?

**Comment by azidar** [comment]
I think it can be backported to 1.2.x?

**Pull request merged**